index.d.ts

```diff
@@ -163,19 +203,57 @@ declare namespace RNTrackPlayer {
     duration: number;
   }
 
-  export class ProgressComponent<P = {}, S = {}> extends Component<P, ProgressComponentState & S> {
+  export class ProgressComponent<P = {}, S = {}> extends Component<
+    P,
+    ProgressComponentState & S
+  > {
     public getProgress: () => number;
     public getBufferedProgress: () => number;
   }
 
   // Constants
-
-  export const STATE_NONE: State;
-  export const STATE_PLAYING: State;
-  export const STATE_PAUSED: State;
-  export const STATE_STOPPED: State;
-  export const STATE_BUFFERING: State;
-  export const STATE_READY: State;
+  export const STATE_NONE: typeof STATE_ANDROID_NONE | typeof STATE_IOS_NONE;
+  export const STATE_PLAYING:
+    | typeof STATE_ANDROID_PLAYING
+    | typeof STATE_IOS_PLAYING;
+  export const STATE_PAUSED:
+    | typeof STATE_ANDROID_PAUSED
+    | typeof STATE_IOS_PAUSED;
+  export const STATE_STOPPED:
+    | typeof STATE_ANDROID_STOPPED
+    | typeof STATE_IOS_STOPPED;
+  export const STATE_CONNECTING:
+    | typeof STATE_ANDROID_CONNECTING
+    | typeof STATE_IOS_CONNECTING;
+  export const STATE_BUFFERING:
+    | typeof STATE_ANDROID_BUFFERING
+    | typeof STATE_IOS_BUFFERING;
+  export const STATE_READY:
+    | typeof STATE_ANDROID_PAUSED
+    | typeof STATE_IOS_READY;
+
+  // Android states: See https://developer.android.com/reference/kotlin/android/support/v4/media/session/PlaybackStateCompat#STATE_CONNECTING:kotlin.Int
+  export const STATE_ANDROID_NONE: 0;
+  export const STATE_ANDROID_STOPPED: 1;
+  export const STATE_ANDROID_PAUSED: 2;
+  export const STATE_ANDROID_PLAYING: 3;
+  export const STATE_ANDROID_FAST_FORWARDING: 4;
+  export const STATE_ANDROID_REWINDING: 5;
+  export const STATE_ANDROID_BUFFERING: 6;
+  export const STATE_ANDROID_ERROR: 7;
+  export const STATE_ANDROID_CONNECTING: 8;
+  export const STATE_ANDROID_SKIPPING_TO_PREVIOUS: 9;
+  export const STATE_ANDROID_SKIPPING_TO_NEXT: 10;
+  export const STATE_ANDROID_SKIPPING_TO_QUEUE_ITEM: 11;
+
+  // iOS states
+  export const STATE_IOS_NONE: "idle";
+  export const STATE_IOS_PLAYING: "playing";
+  export const STATE_IOS_PAUSED: "paused";
+  export const STATE_IOS_STOPPED: "idle";
+  export const STATE_IOS_CONNECTING: undefined;
+  export const STATE_IOS_BUFFERING: "loading";
+  export const STATE_IOS_READY: "ready";
```

Comment:
These constants are defined by the native code just to prevent them being hardcoded here, I think constants should be their own type, not even assuming that they are going to be numbers or strings